cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/SnapshotDefinition.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.table;
+
+
+import co.cask.cdap.api.common.Bytes;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.DatasetAdmin;
+import co.cask.cdap.api.dataset.DatasetContext;
+import co.cask.cdap.api.dataset.DatasetDefinition;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.DatasetSpecification;
+import co.cask.cdap.api.dataset.lib.AbstractDatasetDefinition;
+import co.cask.cdap.api.dataset.lib.CompositeDatasetAdmin;
+import co.cask.cdap.api.dataset.lib.SnapshotDataset;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.tephra.TransactionAware;
+import co.cask.tephra.TransactionExecutor;
+import co.cask.tephra.TransactionExecutorFactory;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ */
+public class SnapshotDefinition extends AbstractDatasetDefinition<SnapshotDataset, DatasetAdmin> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnapshotDefinition.class);
+
+  private static final String METADATA_TABLE_NAME = "metadata";
+  private static final String MAIN_TABLE_NAME = "maindata";
+  private static final String METADATA_PROPERTY_SCHEMA = Schema.recordOf(
+    "record",
+    Schema.Field.of("version", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("value", Schema.of(Schema.Type.LONG))).toString();
+  private static final String METADATA_PROPERTY_ROW_FIELD = "version";
```

Comment:
we seem to have defined these two fields (one as a byte array, the other as a string) two times. Can it be unified?